examples/python3/useful/header.py

```diff
@@ -78,20 +78,26 @@ def addHeader(parentNode, multithreading=False, inverse=False, isConstrained=Fal
                                  multithreading=multithreading, epsilon=1)
         else:
             parentNode.addObject('GenericConstraintSolver', name='ConstraintSolver', tolerance=1e-8, maxIterations=100,
-                                 multithreading=multithreading)
+                                 multithreading=multithreading, printLog=1)
 
     if isContact:
-        contactHeader(parentNode)
+        contactHeader(parentNode, _contact_params=params.contactParams)
 
 
 # components needed for contact modeling
-def contactHeader(parentNode):
+def contactHeader(parentNode, _contact_params=None):
     parentNode.addObject('CollisionPipeline')
     parentNode.addObject("DefaultVisualManagerLoop")
-    parentNode.addObject('RuleBasedContactManager', responseParams='mu=0.8', response='FrictionContactConstraint')
     parentNode.addObject('BruteForceBroadPhase')
     parentNode.addObject('BVHNarrowPhase')
-    parentNode.addObject('LocalMinDistance', alarmDistance=0.05, contactDistance=0.01)
+    if not _contact_params == None:
+        parentNode.addObject('RuleBasedContactManager', responseParams=_contact_params.responseParams,
+                             response='FrictionContactConstraint')
+        parentNode.addObject('LocalMinDistance',  alarmDistance=_contact_params.alarmDistance,
+                             contactDistance=_contact_params.contactDistance)
+    else :
+        parentNode.addObject('RuleBasedContactManager', responseParams='mu=0.1', response='FrictionContactConstraint')
+        parentNode.addObject('LocalMinDistance',  alarmDistance=0.05, contactDistance=0.01)
```

Comment:
The two branch on the conditional are actually doing exactly the same things (but with different values) in parameter). It is better to factorize the code to make it clearer and emphasize the only varing part. 

Something without if/else is better:
```python
def contactHeader(parentNode, _contact_params = DefaultContactParams):
       ...
        parentNode.addObject('RuleBasedContactManager', responseParams=_contact_params.responseParams,
                             response='FrictionContactConstraint')
        parentNode.addObject('LocalMinDistance',  alarmDistance=_contact_params.alarmDistance,
                             contactDistance=_contact_params.contactDistance)
```